### PR TITLE
Add COBS library and tests

### PR DESCRIPTION
**Pull request by Jeff-Ciesielski**
The tests include most of the examples from Wikipedia, as well as some
extras (large packets + failures).

@shonfeder @knitHacker

### CODE REVIEW

File: tests/test.nims [hunk 1]
```diff
@@ -0,0 +1 @@
+switch("path", "$projectDir/../src")
```
Comment (shonfeder) [inline tests/test.nims]: I guess its worth banishing that glyph complaining about a missing newline.

**Review by shonfeder — COMMENTED** [review]
Looks good to me, besides the one thing github is pointing out.

I'll probably pester you for some elucidation of the decode function tomorrow.

File: tests/test.nims (hunk 1 above)
Comment (Jeff-Ciesielski) [reply tests/test.nims]: Fixed

**Comment by shonfeder** [comment]
LGTM

**Pull request merged**